Build failure

We have a dependency on serde 1.0 so we moved to 1.0.95 and now, we have this issue in our build for Android (It builds for other platforms) : 

```
  error[E0432]: unresolved imports `std::sync::atomic::AtomicI64`, `std::sync::atomic::AtomicU64`
     --> /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.95/src/lib.rs:221:33
  
  221 |     pub use std::sync::atomic::{AtomicI64, AtomicU64};
                                   ^^^^^^^^^  ^^^^^^^^^ no `AtomicU64` in `sync::atomic`
                                   |
                                   no `AtomicI64` in `sync::atomic`
  help: a similar name exists in the module
  
  221 |     pub use std::sync::atomic::{AtomicI8, AtomicU64};
                                   ^^^^^^^^
  help: a similar name exists in the module
  
  221 |     pub use std::sync::atomic::{AtomicI64, AtomicU8};
                                              ^^^^^^^^
  
  error: aborting due to previous error

```

I thought I had an old version of rust, but I'm using 1.35.0 so I don't understand why AtomicI64 is not found. Any hint ? Thanks for your help

--- a/serde/build.rs
+++ b/serde/build.rs
@@ -110,7 +110,7 @@ fn target_has_at_least_atomic_u64(target: &str) -> bool {
     // so this data comes from the  src/librustc_target/spec/*.rs
     // files in the rust source. Generally, it's 64-bit platforms
     // plus i686.
-    if target.starts_with("x86-64") || target.starts_with("i686") ||
+    if target.starts_with("x86_64") || target.starts_with("i686") ||
         target.starts_with("aarch64") || target.starts_with("powerpc64") ||
         target.starts_with("sparc64") || target.starts_with("mips64el") {
         true

--- a/serde/build.rs
+++ b/serde/build.rs
@@ -14,6 +14,8 @@ fn main() {
     let target = env::var("TARGET").unwrap();
     let emscripten = target == "asmjs-unknown-emscripten" || target == "wasm32-unknown-emscripten";
 
+    let has_atomic_integers = target_has_at_least_atomic_u64(&target);
+
     // std::collections::Bound was stabilized in Rust 1.17
     // but it was moved to core::ops later in Rust 1.26:
     // https://doc.rust-lang.org/core/ops/enum.Bound.html
@@ -69,7 +71,7 @@ fn main() {
         println!("cargo:rustc-cfg=num_nonzero");
     }
 
-    if minor >= 34 {
+    if minor >= 34 && has_atomic_integers {
         println!("cargo:rustc-cfg=std_integer_atomics");
     }
 }
@@ -101,4 +103,18 @@ fn rustc_minor_version() -> Option<u32> {
     };
 
     u32::from_str(next).ok()
+}
+
+fn target_has_at_least_atomic_u64(target: &str) -> bool {
+    // The cfg variable target_has_atomic is unstable
+    // so this data comes from the  src/librustc_target/spec/*.rs
+    // files in the rust source. Generally, it's 64-bit platforms
+    // plus i686.
+    if target.starts_with("x86-64") || target.starts_with("i686") ||
+        target.starts_with("aarch64") || target.starts_with("powerpc64") ||
+        target.starts_with("sparc64") || target.starts_with("mips64el") {
+        true
+    } else {
+        false
+    }
 }

